blockade/core.py

```diff
@@ -327,15 +337,17 @@ def kill(self, container_names, signal="SIGKILL"):
     def _kill(self, container, signal):
         self.docker_client.kill(container.container_id, signal)
 
-    def stop(self, container_names):
-        containers = self._get_running_containers(container_names)
+    def stop(self, container_names, select_random=False):
+        containers = self._get_running_containers(container_names, select_random)
         for container in containers:
             self._stop(container)
 
     def _stop(self, container):
         self.docker_client.stop(container.container_id, timeout=DEFAULT_KILL_TIMEOUT)
 
-    def start(self, container_names):
+    def start(self, container_names, select_random=False):
+        containers = self._get_stopped_containers(container_names, select_random)
```

Comment:
I think the REST integration test test_action_start fails because it doesn't stop any containers before trying to "start" an UP container (which is ok behavior to expect I think). This is now only looking for the container to be stopped. I think it should work if the start command also gets running containers as well as down.